CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/clean/SplitNodesPlugin.java

```diff
@@ -152,6 +152,25 @@ private void editResultStore(final RecordStore result, final String left, final
             result.set(GraphRecordStoreUtilities.SOURCE + AnalyticConcept.VertexAttribute.TYPE, leftVertexTypesMatches.get(0));
         }
         result.set(GraphRecordStoreUtilities.DESTINATION + VisualConcept.VertexAttribute.IDENTIFIER, right);
+        
+        //Loops through all of the Node attributes and copies them to the new node
+        query.reset();
+        while(query.next()){
+            for (final String key : query.keys()){
+                if(SOURCE_ID.replace("<string>", "").equals(key) || SOURCE_IDENTIFIER.replace("<string>", "").equals(key)) {
```

Comment:
@formalhaut69 thanks for this.

@cygnus-x-1 I wanted to ask if you have any ideas about how we could format this a bit better so that we don't have to be so verbose and replace the type `<string>` to get this to work.

We can replace `query.keys()` with `((GraphRecordStore) query).keysWithType()` which will work for line 160 but then the if block (162-164) comparing x, y and z will not work because the key is now `x<float>`.

To me something looks wrong with the design of the `GraphRecordStore`. Keen for feedback from anyone.

Reply:
You shouldn't need to worry about the field type, as the default behaviour of GraphRecordStore.keys is to explicitly **not** return field types.

Reply:
https://github.com/constellation-app/constellation/blob/28443a95682f6240cfca4d4792ca13000f3fdbf3/CoreGraph/src/au/gov/asd/tac/constellation/graph/processing/GraphRecordStoreUtilities.java#L59
@cygnus-x-1 @arcturus2 I've made a mistake here sorry, the SOURCE_IDENTIFIER never contains ```<string>``` so when comparing it to the key, we don't need to replace ```<string>```. The SOURCE_ID does still contain it's type so we still need to replace ```<string>``` for it. You can see in GraphRecordStoreUtilities it's declared with ```<string>``` in its value . The X,Y and Z values are declared without their type.

Reply:
While `GraphRecordStoreUtilites.ID` does explicitly declare a type (because it's designed to work with `GraphRecordStore` which requires typed fields), this type will only be recorded in the `typedRecords` object. `GraphRecordStore.keys` will return the keys of its `records` object (which has no types as it comes from `TabularRecordStore`) whereas `GraphRecordStore.keysWithType` will return keys from `typedRecords`. Since you are using `GraphRecordStore.keys` in this case, you shouldn't have to worry about types.

Reply:
Thanks @formalhaut69 for the fix.

Thanks @cygnus-x-1 for the explanation.

It looks like we are using features from the `TabularRecordStore` and `GraphRecordStore` causing issues. Though I'm not a fan of magic strings (i.e ".[id]"), I don't know how best to solve this one without going deep into the framework...and I don't feel its worth it right now given what @formalhaut69 has proposed works and I'm worrying about semantics.

Reply:
A simple fix might be to create ```TabularRecordStoreUtilities.ID``` as an analogue to ```GraphRecordStoreUtilities.ID``` and to use that instead.